cmake/modules/Libdispatch.cmake

```diff
@@ -77,6 +77,8 @@ foreach(sdk ${DISPATCH_SDKS})
                           -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}
                           -DCMAKE_SYSTEM_NAME=${SWIFT_SDK_${sdk}_NAME}
                           -DCMAKE_SYSTEM_PROCESSOR=${arch}
+                          "-DCMAKE_ANDROID_NDK=${SWIFT_ANDROID_NDK_PATH}"
+                          -DCMAKE_ANDROID_ARCH_ABI=${SWIFT_SDK_ANDROID_ARCH_${arch}_ABI}
                           -DBUILD_SHARED_LIBS=YES
                           -DENABLE_SWIFT=NO
                           -DENABLE_TESTING=NO
```

Comment:
I think `INSTALL_DIR` ends up being `<PREFIX>` for every platform, so every SDK/arch will overwrite each others libdispatch. I think `INSTALL_DIR ${CMAKE_BINARY_DIR}/lib/swift/${SWIFT_SDK_${sdk}_LIB_SUBDIR}/${arch}` or similar is needed in the configuration of `ExternalProject_Add`

Since this is used also for the host, and those were installed directly into the `<PREFIX>/lib`, there might need a way of switching between the two formats, but I am not sure (maybe @compnerd knows how to approach this).

Reply:
The `INSTALL_DIR` includes SDK and arch. On Linux x86_64 it is `[...]/libdispatch-linux-x86_64-prefix`.

Reply:
IIRC, the `CMAKE_INSTALL_PREFIX=<INSTALL_DIR>` results in the installation into the target's name, so it should install to `libdispatch-${VARIANT_NAME}-prefix`.

Reply:
@compnerd: yes, you are right, but I was not seeing that. The problem is that the configuration was being executed twice, and the second time the values passed originally were lost. See also my explanation in  https://github.com/apple/swift/pull/35881#pullrequestreview-588089265